### ISSUE TITLE
Enhancement Request: Argument Passing 

### ISSUE THREAD

**Issue by ghost** [nikclayton/ob-sql-mode#13]
As compared to the back-end ob-sql, I miss the ability to pass argument values to source blocks.

For instance, this does not work:
```
#+begin_src sql-mode :var myarg=7.8
  SELECT $myarg + 11 AS sum;
#+end_src
```
I have only tried PostgreSQL this far, but judging from the error messages, it is the argument passing which fails.

**Comment by bigodel** [comment]
I second this. There is [a comment](https://github.com/nikclayton/ob-sql-mode/blob/2eaf436a6ac2178b94442d80f84fc6c02aa644d8/ob-sql-mode.el#L125) on the code saying the author doesn't have a use for it, but I use it extensively to reuse query results. I could be doing that some other way, but I think it is reasonable for a Org Babel back-end to support this. I could probably take a stab at it, but I won't have time available for the foreseeable months...

**Comment by bigodel** [comment]
I would also like to add that it wouldn't be much trouble to support passing user, password, database and such as header arguments, while still letting `sql.el` handle it (referencing [this other comment](https://github.com/nikclayton/ob-sql-mode/blob/2eaf436a6ac2178b94442d80f84fc6c02aa644d8/ob-sql-mode.el#L135)) just like its being done with `sql-product`, we'd just have to get these arguments and bind the appropriate variables (i.e. `sql-user`, `sql-password`, `sql-database`, ...)

**Comment by flintforge** [comment]
The two cases suggested in this issue — variables (:var) and  arguments — e.g. header-arguments for connection — are covered by [ob-sql-session](https://github.com/flintforge/ob-sql-session). Regarding the last comment, please note that while org variables are, for some programming languages, a simple matter of substitution/evaluation, and as such a piece of cake, arguments, on the other hand, requires careful handling.

**Comment by bigodel** [comment]
Yeah, at the time of writing this comment ob-sql-session wasn't a thing but I agree it covers the cases mentioned
